### ISSUE TITLE
Announce plugin should use system notifications instead of Growl on Mac OS

### ISSUE THREAD

**Issue by wolfs** [gradle/gradle#739]
### Expected Behavior

When using the Announce plugin on Mac OS we should use system notifications.
### Current Behavior

We try to use Growl via Apple script which does not if Growl is not installed and makes the build fail. This also makes tests in `:announce` fail and as a consequence runnning `gradlew build` for the build tool itself results in an error.
### Steps to Reproduce (for bugs)

Run `./gradlew build` on Mac OS. Then `BuildAnnouncementsPluginIntegrationTest` will fail.
### Your Environment
- Build scan URL: https://scans.gradle.com/s/acurmrrynyewa
### Implementation

Using system notification via Apple script seem to be [easy](https://macosxautomation.com/mavericks/notifications/01.html). We only would need to change [AppleScriptBackedGrowlAnnouncer](https://github.com/gradle/gradle/blob/b072033c739c3fcf0eb469dcc7cce9d688f642d5/subprojects/announce/src/main/java/org/gradle/api/plugins/announce/internal/jdk6/AppleScriptBackedGrowlAnnouncer.java#L48-L62).

**Comment by huxi** [comment]
I tried implementing something like that a while back and it really only _seems_ to be easy.

**TL;DR** It used to be possible to execute AppleScript via `javax.script.*` in Java 6 but that doesn't work for Java 7 and Java 8 anymore. There is no way of getting this to work for both Java 6 and Java 7/8 that I'm aware of.

The only compatible way would need to execute `/usr/bin/osascript` but those notifications would always have the generic Script Editor icons.

I ultimately gave up in frustration. Hope this helps anyway.

Here are some helpful links:
- [Related AppleScript documentation](https://developer.apple.com/library/content/documentation/AppleScript/Conceptual/AppleScriptLangGuide/reference/ASLR_cmds.html#//apple_ref/doc/uid/TP40000983-CH216-SW224)
- [How to execute AppleScript from a Java or Scala application](http://alvinalexander.com/scala/how-to-execute-applescript-from-java-scala)
- [How to configure AppleScript to work with Java 7 on Mac OS X](http://alvinalexander.com/photos/using-applescript-java-7-mac-os-x-109)
- [How to fire Mac OS X notifications with AppleScript (and Scala)](http://alvinalexander.com/mac-os-x/how-to-fire-mac-notifications-applescript-scala-java)

```groovy
// This Groovy script should work but won't
import javax.script.*

def manager = new ScriptEngineManager()
def engine = manager.getEngineByName('AppleScript')
println engine // will be null

def user = 'world'
def notificationCommand = "display notification \"Hello, $user!\" subtitle \"Some subtitle.\""
println notificationCommand

engine.eval notificationCommand
```

**Comment by adrienlenag** [comment]
Hi there,

I have the exact same problem with MacOS X Sierra, and gradle 2.14.1, compiling using java8 (jdk 1.8.0_74 to be precise)
Here is what I get when I want to generate specific resources with gradle :

```
Caused by: javax.script.ScriptException: Can’t get application id "com.Growl.GrowlHelperApp".
        at apple.applescript.AppleScriptEngine.evalScript(Native Method)
        at apple.applescript.AppleScriptEngine.eval(AppleScriptEngine.java:368)
        at apple.applescript.AppleScriptEngine.eval(AppleScriptEngine.java:339)
        at org.gradle.api.plugins.announce.internal.jdk6.AppleScriptBackedGrowlAnnouncer.send(AppleScriptBackedGrowlAnnouncer.java:63)
        at org.gradle.api.plugins.announce.internal.IgnoreUnavailableAnnouncer.send(IgnoreUnavailableAnnouncer.java:34)
        at org.gradle.api.plugins.announce.AnnouncePluginExtension$LocalAnnouncer.send(AnnouncePluginExtension.java:119)
```

Is there a way to tell gradle not to use Growl for notifications, but use Apple native one instead ?
Am really stuck here as I cannot generate my Grammar source

Thanks for your help

**Comment by eskatos** [comment]
This plugin actually fails the build during configuration time. It should at least be moved to execution time, and maybe only warn, or provide a flag to ignore failures.

**Comment by jjohannes** [comment]
The _announce_ plugin is removed in Gradle 6.0.
